commons/all/src/main/java/org/infinispan/commons/util/EmptyIntSet.java

```diff
@@ -6,15 +6,25 @@
 import java.util.Set;
 import java.util.stream.IntStream;
 
+import org.infinispan.commons.marshall.ProtoStreamTypeIds;
+import org.infinispan.protostream.annotations.ProtoFactory;
+import org.infinispan.protostream.annotations.ProtoTypeId;
+
 /**
  * Immutable empty IntSet
  * @author wburns
  * @since 9.3
  */
-class EmptyIntSet extends AbstractImmutableIntSet {
+@ProtoTypeId(ProtoStreamTypeIds.INTSET_EMPTY)
+public class EmptyIntSet extends AbstractImmutableIntSet {
    // We just use default hashCode as we only have 1 instance
    private final static EmptyIntSet INSTANCE = new EmptyIntSet();
 
+   @ProtoFactory
```

Comment:
can't use `getInstance()`?

Reply:
:+1: If we change the return type to be the concrete implementation we can.